lib/sensu/transport/snssqs.rb

```diff
@@ -173,7 +184,19 @@ def receive_messages
             wait_time_seconds: @settings[:wait_time_seconds],
             max_number_of_messages: @settings[:max_number_of_messages],
           )
-          resp.messages
+          resp.messages.map do |msg|
+            if !msg.key? 'message_attributes'
```

Comment:
A message without the message_attributes directly in the root is an encapsulated SNS message (JSON formated as string), it therefore needs two changes:
1. The message_attributes need to be extracted from the body
2. The body needs to be replaced with the original message.